From 8c7a9a0442d304851350e80e990cd3a05e4b8a0b Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Wed, 6 Sep 2023 15:40:03 +0800
Subject: [PATCH 191/232] va: baseenc: use in_info rather than
 input_state->info

The in_info is correct video info which already considers the
DMA kind caps case.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5289>
---
 sys/va/gstvabaseenc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sys/va/gstvabaseenc.c b/sys/va/gstvabaseenc.c
index 5b2d84f96..15f0d7655 100644
--- a/sys/va/gstvabaseenc.c
+++ b/sys/va/gstvabaseenc.c
@@ -219,7 +219,7 @@ _get_sinkpad_pool (GstElement * element, gpointer data)
 
   gst_allocation_params_init (&params);
 
-  size = GST_VIDEO_INFO_SIZE (&base->input_state->info);
+  size = GST_VIDEO_INFO_SIZE (&base->in_info);
 
   surface_formats = gst_va_encoder_get_surface_formats (base->encoder);
 
@@ -261,7 +261,7 @@ gst_va_base_enc_import_input_buffer (GstVaBaseEnc * base,
 #endif
     .display = base->display,
     .entrypoint = GST_VA_BASE_ENC_ENTRYPOINT (base),
-    .in_info = &base->input_state->info,
+    .in_info = &base->in_info,
     .sinkpad_info = &base->priv->sinkpad_info,
     .get_sinkpad_pool = _get_sinkpad_pool,
   };
@@ -975,8 +975,8 @@ gst_va_base_enc_add_frame_rate_parameter (GstVaBaseEnc * base,
     /* denominator = framerate >> 16 & 0xffff;
      * numerator   = framerate & 0xffff; */
     .fr.framerate =
-        (GST_VIDEO_INFO_FPS_N (&base->input_state->info) & 0xffff) |
-        ((GST_VIDEO_INFO_FPS_D (&base->input_state->info) & 0xffff) << 16)
+        (GST_VIDEO_INFO_FPS_N (&base->in_info) & 0xffff) |
+        ((GST_VIDEO_INFO_FPS_D (&base->in_info) & 0xffff) << 16)
   };
   /* *INDENT-ON* */
 
-- 
2.40.1

